src/ui/layout/RecentObjectsList.vue

```diff
@@ -0,0 +1,200 @@
+            :object-path="recentObject.objectPath"
+            :navigation-path="recentObject.navigationPath"
+            :domain-object="recentObject.domainObject"
+            @openAndScrollTo="openAndScrollTo($event)"
+        />
+    </ul>
+</div>
+</template>
+
+<script>
+const MAX_RECENT_ITEMS = 20;
+const LOCAL_STORAGE_KEY__RECENT_OBJECTS = 'mct-recent-objects';
+import RecentObjectsListItem from './RecentObjectsListItem.vue';
+export default {
+    name: 'RecentObjectsList',
+    components: {
+        RecentObjectsListItem
+    },
+    inject: ['openmct'],
+    props: {
+
+    },
+    data() {
+        return {
+            recents: []
+        };
+    },
+    computed: {
+        recentObjects() {
+            return this.recents.filter((recentObject) => {
+                return recentObject.location !== null;
+            });
+        }
+    },
+    mounted() {
+        this.compositionCollections = {};
+        this.openmct.router.on('change:path', this.onPathChange);
+        this.getSavedRecentItems();
+    },
+    destroyed() {
+        this.openmct.router.off('change:path', this.onPathChange);
+    },
+    methods: {
+        /**
+         * Add a composition collection to the map and register its remove handler
+         * @param {string} navigationPath
+         */
+        addCompositionListenerFor(navigationPath) {
+            this.compositionCollections[navigationPath].removeHandler = this.compositionRemoveHandler(navigationPath);
+            this.compositionCollections[navigationPath].collection.on('remove',
+                this.compositionCollections[navigationPath].removeHandler);
+        },
+        /**
+         * Handler for composition collection remove events.
+         * Removes the object and any of its children from the recents list.
+         * @param {string} navigationPath
+         */
+        compositionRemoveHandler(navigationPath) {
+            /**
+             * @param {import('../../api/objects/ObjectAPI').Identifier | string} identifier
+             */
+            return (identifier) => {
+                // Construct the navigationPath of the removed object itself
+                const removedNavigationPath = `${navigationPath}/${this.openmct.objects.makeKeyString(identifier)}`;
+
+                // Remove the object and any of its children from the recents list
+                this.recents = this.recents.filter((recentObject) => {
+                    return !recentObject.navigationPath.includes(removedNavigationPath);
+                });
+
+                this.removeCompositionListenerFor(removedNavigationPath);
+            };
+        },
+        /**
+         * Restores the RecentObjects list from localStorage, retrieves composition collections,
+         * and registers composition listeners for composable objects.
+         */
+        getSavedRecentItems() {
+            const savedRecentsString = localStorage.getItem(LOCAL_STORAGE_KEY__RECENT_OBJECTS);
```

Comment:
Do we want to move any of this to the ObjectAPI?

Reply:
Good question. We save/load state to localStorage in the same way in mct-tree, so this is certainly mostly boilerplate that can be extracted. I've heard some discussion about revamping localStorage management, not sure if ObjectAPI would be the place though. @akhenry I'm deflecting to you, thoughts?

Either way, feels like something that would warrant its own separate issue/PR

Reply:
Agree about a separate PR, was mainly curious about architecture here.